### PR TITLE
Enable log upload for unit and verification tests

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov ptal

File: tools/release-ci/cmd/sidecar.go [hunk 1]
```diff
@@ -55,7 +61,7 @@ func runSidecar(_ []string) error {
 		return err
 	}
 
-	if len(config.GceCredentialsFile) == 0 {
+	if len(config.GceCredentialsFile) == 0 || !exists(config.GceCredentialsFile) {
```
Comment (stevekuznetsov) [inline tools/release-ci/cmd/sidecar.go]: Seems like specifying a file that doesn't exist is an error, not something to eat and continue

File: tools/release-ci/cmd/sidecar.go [hunk 2]
```diff
@@ -100,31 +106,38 @@ func retrieveProcessReturnCode(markerFile string) (int, error) {
...
-				if event.Name == markerFile && event.Op&fsnotify.Create == fsnotify.Create {
-					break
+
+	// Only start watching file events if the file doesn't exist
+	// If the file exists, it means the main process already completed.
+	if !exists(markerFile) {
+		group.Add(1)
+		go func() {
+			defer group.Done()
+		outerLoop:
```
Comment (stevekuznetsov) [inline tools/release-ci/cmd/sidecar.go]: Does a refactor without labels look uglier?

File: tools/release-ci/pkg/config/batch.go [hunk 3]
```diff
@@ -41,14 +44,15 @@ type Repo struct {
...
+	RepoMeta
 	Repo
 }
 
 // GcsPath is the base path for job data. Batch jobs
 // store their information only in the pull request
 // directory, not for any specific PR
 func (b *Batch) GcsPath() string {
-	return fmt.Sprintf("%s/%s/%d/", PullRequestDirectory, b.JobName, b.BuildNumber)
+	return fmt.Sprintf("%s/%s/%s/%s/%d/", PullRequestDirectory, b.RepoOwner, b.RepoName, b.JobName, b.BuildNumber)
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/config/batch.go]: Convention is `org_repo` not `org/repo` -- I don't like it but not sure churn to make it cleaner is worth it

File: tools/release-ci/pkg/config/save.go [hunk 4]
```diff
@@ -105,6 +113,19 @@ func retrieveRepoConfig() (Repo, bool, error) {
 		missing = append(missing, "REPO_NAME")
 	}
 
+	if len(missing) > 0 {
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/config/save.go]: These are not going to be set for periodics, so this is not always an error

File: tools/release-ci/pkg/logging/gcs/finished.go [hunk 5]
```diff
@@ -35,19 +36,21 @@ func UploadFinishedData(processLog, metadataFile, artifactDir string, passed boo
 
 	gcsPath := metadata.GcsPath()
 	uploadTargets := map[string]uploadFunc{
-		gcsPath + "/finished.json": dataUpload(finishedData),
+		ospath.Join(gcsPath, "finished.json"): dataUpload(finishedData),
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/logging/gcs/finished.go]: I hope to see the day we run this in a Windows container ;)

Comment (csrwng) [reply tools/release-ci/pkg/logging/gcs/finished.go]: it's not that :) ... actually path.Join will always use "/". The problem is we were getting duplicate "/"s

File: tools/release-ci/cmd/sidecar.go (hunk 2 above)
Comment (csrwng) [reply tools/release-ci/cmd/sidecar.go]: No, I can do that. I was not thinking of making it pretty, but as it was, it never exited the loop

File: tools/release-ci/pkg/config/batch.go (hunk 3 above)
Comment (csrwng) [reply tools/release-ci/pkg/config/batch.go]: Will fix

File: tools/release-ci/pkg/config/save.go (hunk 4 above)
Comment (csrwng) [reply tools/release-ci/pkg/config/save.go]: Our pipeline will always set it

**Comment by csrwng** [comment]
comments addressed :)

**Pull request merged**